### DISCUSSION TITLE
da_core_news_md is not loading on my server

### DISCUSSION THREAD

**Discussion by Blaabjerg96** [explosion/spaCy#7997] [Help: Coding & Implementations] [answered]
Hello I've created a AWS server and connected RASA X to it. Afterwards I connected my bot through github successfully. 
The problem arises, when I try to talk to my bot. It says that the model that I use is not a spaCy model.

And refers to say I should download it, or link it. 

But I make the download statement in the docker file, but am I using the model incorrectly or what is going wrong? 

Thanks in advance

EDIT: Solved - the model was wrongly uploaded to the RASA x server on AWS

**Comment by svlandeg** [comment, marked as answer]
Hi!

It's very difficult to say what may be going wrong without some more information. What are your install commands for spaCy and for the Danish model? Were there any warnings during install/download? What is the error message you're getting when trying to use this model? Can you have a look at the `pip list` output of the virtual environment you're running this in, and verify whether or not `da_core_news_md` is in that list?

**Reply by Blaabjerg96** [reply]
Sorry for the late reply. 

I managed to download spacy and the danish model on the AWS through the command interface. The problem is, that my RASA assistant still cant seem to find the model that im downloading. 

Am I missing something? The idea is that I would like to run my RASA X on AWS, and connect my bot via github. Should the model file be on github?

**Reply by polm** [reply]
Not familiar with RASA on AWS, but if you want help with this you need to provide:

1. The commands you have run to install install spaCy and the model (and how your do it in your RASA assistant)
2. The error messages you are getting

When you say "my RASA assistant still cant seem to find the model" it's not clear what that means.

**Reply by Blaabjerg96** [reply]
Locally I created a virtual enviroment, where I ran:
Pip install Spacy
python -m download spacy da_core_news_md

On the server i just ran those commands without the enviroment.

The error message i get is:
InvalidModelError: Model 'da_core_news_md' is not a linked spaCy model

Sorry for the ambiguous comments 

edit: the actual error message is:

`During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/agent.py", line 161, in _update_model_from_server
    _load_and_set_updated_model(agent, model_directory, new_fingerprint)
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/agent.py", line 134, in _load_and_set_updated_model
    interpreter = _load_interpreter(agent, nlu_path)
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/agent.py", line 93, in _load_interpreter
    return rasa.core.interpreter.create_interpreter(nlu_path)
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/interpreter.py", line 33, in create_interpreter
    return RasaNLUInterpreter(model_directory=obj)
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/interpreter.py", line 127, in __init__
    self._load_interpreter()
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/interpreter.py", line 165, in _load_interpreter
    self.interpreter = Interpreter.load(self.model_directory)
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/model.py", line 331, in load
    return Interpreter.create(
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/model.py", line 405, in create
    component = component_builder.load_component(
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/components.py", line 824, in load_component
    component = registry.load_component_by_meta(
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/registry.py", line 177, in load_component_by_meta
    return component_class.load(
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/utils/spacy_utils.py", line 309, in load
    nlp = cls.load_model(model_name)
  File "/opt/venv/lib/python3.8/site-packages/rasa/nlu/utils/spacy_utils.py", line 52, in load_model
    raise InvalidModelError(
rasa.nlu.model.InvalidModelError: Please confirm that da_core_news_md is an available spaCy model. You need to download one upfront. For example:
python -m spacy download en_core_web_md
More informaton can be found on https://rasa.com/docs/rasa/components#spacynlp`


Thanks in advance

**Reply by polm** [reply]
That error is coming from inside Rasa, so it looks like it might be an issue with their spaCy utils and you should ask at their repo. 

That said, one other thing you could test that would be helpful is whether you can install the English models. I think the Danish models have been around for a while so it shouldn't make a difference, but if English work but Danish doesn't that might be a useful hint.

**Reply by Blaabjerg96** [reply]
Thanks a lot for the help I will open an issue over at Rasa's github.

**Reply by polm** [reply]
Sorry we couldn't be more help. Please do link from this thread to the issue over there after you open it, in case anyone stumbles in here through a search.

**Reply by Blaabjerg96** [reply]
Not necessary - I found the error - Apparantly the model was uploaded incorrectly to my Rasa X assistant. Thanks alot for the help

**Reply by svlandeg** [reply]
Happy to hear you found the problem, thanks for reporting back!
